### PR TITLE
fix(windows): stop console window flashes

### PR DESCRIPTION
**Pull request by danyaberezun**
## Summary

Prevent Windows console flashes by setting `windowsHide: true` on background child processes. Bun 1.4.0 maps this to `UV_PROCESS_WINDOWS_HIDE`.

Adds shared sync/detached spawn helpers and covers git refresh, terminal checks, bounded subprocesses, Central, dialogs, editors, browser launch, PowerShell, and trash helpers.

## Checklist

- [x] Fast gates pass: `bun run lint`, `bun run typecheck`, `bun run test`
- [x] E2E suite passes for app-affecting changes (`bun run e2e`, or `bun run e2e:full` when touching agent behavior)
- [x] Relevant `SPEC.md` / top-level specs updated to reflect any boundary, contract, or behavior change
- [x] I have read the [Contributing guide](../CONTRIBUTING.md) and agree to the [Code of Conduct](../CODE_OF_CONDUCT.md)

### CODE REVIEW

File: packages/shared/src/spawn.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		stdout: (result.stdout ?? Buffer.alloc(0)).toString("utf8"),
+		stderr: (result.stderr ?? Buffer.alloc(0)).toString("utf8"),
+	};
+}
+
+export function spawnDetached(argv: readonly string[], options: SpawnEnvironment = {}): boolean {
+	const [command, ...args] = argv;
+	if (!command) return false;
+	try {
+		const child = spawn(command, args, {
```
[the remark this answers is not in the document]
Comment (danyaberezun) [reply packages/shared/src/spawn.ts]: Fixed in 85731a9: spawnDetached now waits for spawn/error, consumes ENOENT, returns false for missing executables, and tests a non-empty missing argv.

File: packages/shared/src/spawn.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	argv: readonly string[],
+	options: SpawnEnvironment & { timeoutMs?: number; maxBuffer?: number } = {},
+): SpawnSyncCaptured {
+	const [command, ...args] = argv;
+	if (!command) return { launched: false, exitCode: null, stdout: "", stderr: "" };
+	const result = spawnSync(command, args, {
+		cwd: options.cwd,
+		env: options.env as NodeJS.ProcessEnv | undefined,
+		windowsHide: true,
+		encoding: "buffer",
```
[the remark this answers is not in the document]
Comment (danyaberezun) [reply packages/shared/src/spawn.ts]: Resolved in 6038f10 by using Bun native spawn directly. Bun 1.4.0 forwards windowsHide in node:child_process (child_process.ts:1442) and native spawn sets UV_PROCESS_WINDOWS_HIDE (process.rs:1823-1824): https://github.com/oven-sh/bun/blob/bun-v1.4.0/src/spawn/process.rs#L1823-L1825

**Review by rsolmano — APPROVED** [review]
